official/resnet/keras/resnet_cifar_model.py

```diff
@@ -0,0 +1,325 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""ResNet56 model for Keras adapted from tf.keras.applications.ResNet50.
+
+# Reference:
+- [Deep Residual Learning for Image Recognition](
+    https://arxiv.org/abs/1512.03385)
+Adapted from code contributed by BigMoyan.
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import warnings
+
+import tensorflow as tf
+
+
+BATCH_NORM_DECAY = 0.997
+BATCH_NORM_EPSILON = 1e-5
+L2_WEIGHT_DECAY = 2e-4
+
+
+def _obtain_input_shape(input_shape,
+                        default_size,
+                        data_format):
+  """Internal utility to compute/validate a model's input shape.
+
+  Arguments:
+    input_shape: Either None (will return the default network input shape),
+        or a user-provided shape to be validated.
+    default_size: Default input width/height for the model.
+    data_format: Image data format to use.
+
+  Returns:
+    An integer shape tuple (may include None entries).
+
+  Raises:
+    ValueError: In case of invalid argument values.
+  """
+  if input_shape and len(input_shape) == 3:
+    if data_format == 'channels_first':
+      if input_shape[0] not in {1, 3}:
+        warnings.warn(
+            'This model usually expects 1 or 3 input channels. '
+            'However, it was passed an input_shape with ' +
+            str(input_shape[0]) + ' input channels.')
+      default_shape = (input_shape[0], default_size, default_size)
+    else:
+      if input_shape[-1] not in {1, 3}:
+        warnings.warn(
+            'This model usually expects 1 or 3 input channels. '
+            'However, it was passed an input_shape with ' +
+            str(input_shape[-1]) + ' input channels.')
+      default_shape = (default_size, default_size, input_shape[-1])
+
+  return input_shape
```

Comment:
Should this return default_shape?  It does not seem like it does anything.  Apologies if I missed you resolving this comment last time.  Github code review is confusing.  We likely can remove this code and just assume CIFAR-10 3 channels with the only option being channels first and last.

Reply:
I delete this method and do it as resnet50 does.